### ISSUE TITLE
Sorting of silent addresses when creating outputs

### ISSUE THREAD

**Issue by Sosthene00** [cygnet3/rust-silentpayments#11]
For now we simply group `m_pubkey`s that have the same `scan_pubkey` in the order we receive them. It means that if you change the order you'll get different keys because of the `_n_` counter.

I think it would be better to have some kind of sorting that ensure that no matter the order we receive the arguments we always produce the same keys in outputs. 

Since labels are `Scalar`s I think it's a no brainer to sort them by ascending order. 

Basically if we want to create a transaction that pays to (A,0), (A,1), (B,101), (B,102) and (C,201), we will always sort the silent addresses in this order before generating the keys no matter the order the caller is sending us the inputs. That would not be the case today.

I don't think this is clear in the BIP for now, and multiple labels test cases fail when I tried to sort the labeled keys so that's definitely something to consider.

**Comment by josibake** [comment]
There is no way for the sender to sort by the label values since the sender has no way of knowing what label values (if any) were used by the recipient.

It is also worth mentioning that the receiver can find the outputs regardless of what order the sender creates them. I do think it would be helpful to specify an order but only for determinism in testing.  Perhaps sorting by the (x,y) coordinates of the point?

**Comment by Sosthene00** [comment]
Yes reading your comment on the BIP made me realise I was confused about the implication for the receiver. 

If this is so I think either letting things as they are now adding a comment to say that order of inputs matter for the test accuracy, or modifying the test vectors to make the receiver check for the label instead of the actual pubkey.
